packages/babel-generator/test/index.js

```diff
@@ -298,6 +298,29 @@ describe("generation", function() {
 });
 
 describe("programmatic generation", function() {
+  it("should add parenthesis when NullishCoalescing is used along with ||", function() {
+    // https://github.com/babel/babel/issues/10260
+    const nullishCoalesc = t.logicalExpression(
+      "??",
+      t.logicalExpression("||", t.identifier("a"), t.identifier("b")),
+      t.identifier("c"),
+    );
+    const output = generate(nullishCoalesc).code;
+    expect(output).toBe(`(a || b) ?? c`);
+  });
+  it("should add parenthesis when NullishCoalesing is used with &&", function() {
+    const nullishCoalesc = t.logicalExpression(
+      "??",
+      t.identifier("a"),
+      t.logicalExpression(
+        "&&",
+        t.identifier("b"),
+        t.logicalExpression("&&", t.identifier("c"), t.identifier("d")),
+      ),
+    );
+    const output = generate(nullishCoalesc).code;
+    expect(output).toBe(`a ?? (b && c && d)`);
+  });
```

Comment:
Nit: Newline

Reply:
my editor settings :( I will remove the new lines added.

Reply:
Seems I mis understood the statement. What you meant was add a new line between the tests. Adding them now.